artman/cli/main2.py

```diff
@@ -0,0 +1,435 @@
+        artifact_config, flags.input_dir, flags.output_dir)
+    logger.debug('Below is the legacy config after conversion:\n%s' %
+                 pprint.pformat(legacy_config_dict))
+    tmp_legacy_config_yaml = '%s.tmp' % artman_config_path
+    with open(tmp_legacy_config_yaml, 'w') as outfile:
+        yaml.dump(legacy_config_dict, outfile, default_flow_style=False)
+
+    googleapis = os.path.realpath(
+        os.path.expanduser(
+            pipeline_args['local_paths']['googleapis'], ))
+    config = ','.join([
+        '{artman_config_path}',
+        '{googleapis}/gapic/lang/{shared_config_name}',
+    ]).format(
+        artman_config_path=tmp_legacy_config_yaml,
+        googleapis=googleapis,
+        shared_config_name=shared_config_name, )
+
+    # Set the pipeline as well as package_type and packaging
+    artifact_type = artifact_config.type
+    if artifact_type == config_pb2.Artifact.GAPIC:
+        pipeline_name = 'GapicClientPipeline'
+    elif artifact_type == config_pb2.Artifact.GAPIC_CONFIG:
+        pipeline_name = 'GapicConfigPipeline'
+    elif artifact_type == config_pb2.Artifact.GRPC:
+        pipeline_name = 'GrpcClientPipeline'
+    elif artifact_type == config_pb2.Artifact.GRPC_COMMON:
+        pipeline_name = 'GrpcClientPipeline'
+    elif artifact_type == config_pb2.Artifact.GAPIC_ONLY:
+        pipeline_name = 'GapicClientPipeline'
+
+    language = config_pb2.Artifact.Language.Name(
+        artifact_config.language).lower()
+    pipeline_args['language'] = language
+
+    # Parse out the full configuration.
+    # Note: the var replacement is still needed because they are still being
+    # used in some shared/common config yamls.
+    config_sections = ['common']
+    for config_spec in config.split(','):
+        config_args = config_util.load_config_spec(
+            config_spec=config_spec,
+            config_sections=config_sections,
+            repl_vars={
+                k.upper(): v
+                for k, v in pipeline_args['local_paths'].items()
+            },
+            language=language, )
+        pipeline_args.update(config_args)
+
+    # Setup publishing related config if needed.
+    if flags.subcommand == 'generate':
+        pipeline_args['publish'] = 'noop'
+    elif flags.subcommand == 'publish':
+        publishing_config = _get_publishing_config(artifact_config,
+                                                   flags.target)
+        if publishing_config.type == config_pb2.Artifact.PublishTarget.GITHUB:
+            pipeline_args['publish'] = 'local' if flags.dryrun else 'github'
+            pipeline_args['github'] = support.parse_github_credentials(
+                argv_flags=flags,
+                config=user_config.get('github', {}), )
+            repos = pipeline_args.pop('git_repos')
+            pipeline_args['git_repo'] = support.select_git_repo(
+                repos, publishing_config.name)
+        else:
+            raise NameError(
```

Comment:
`ValueError` or `RuntimeError`. There is a very specific meaning for `NameError` (use of an undefined variable) which is not correct here.